storm-client/src/jvm/org/apache/storm/metric/FileBasedEventLogger.java

```diff
@@ -59,8 +62,12 @@ private void initLogWriter(Path logFilePath) {
 
 
     private void setUpFlushTask() {
-        ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
-        Runnable task = new Runnable() {
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("event-logger-flush-%d")
```

Comment:
should we add `setDaemon(true)` ?

Reply:
By default daemon is set as false, it should be set as true.

Reply:
@arunmahadevan @satishd 

I intended to set daemon as false, since I would like to provide a chance to shutdown gracefully, via to wait for current running task to be finished instead of killing immediately.
Typically `shutdownNow()` will trigger interrupt and run() method looks like be able to handle interrupt (though it will throw RuntimeException).

If we are in favor of guaranteeing JVM shutdown more than graceful shutdown, I'm happy to  change it.

Reply:
I'm in favor of guaranteeing JVM shutdown in this case as I have seen dangling worker scenarios are hard for users to debug and maintain.

Reply:
OK, makes sense. Will add `setDaemon(true)`.

Reply:
Addressed.